<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QAvr</class>
 <widget class="QWidget" name="QAvr">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>638</width>
    <height>575</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QAvr</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="margin">
    <number>2</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tab_main">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Programmer</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>MCU:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="mcu">
           <property name="editable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QSplitter" name="flash_output_splitter">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <widget class="QTabWidget" name="programmer_tabs">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_4">
           <attribute name="title">
            <string>Flash</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0" colspan="4">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>flash file:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="flash_file">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="flash_format"/>
              </item>
              <item>
               <widget class="QPushButton" name="select_hex_file">
                <property name="text">
                 <string>Select</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="hex_upload">
              <property name="text">
               <string>Upload</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QPushButton" name="load_flash">
              <property name="text">
               <string>Read</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QPushButton" name="save_flash">
              <property name="text">
               <string>Save to...</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QPushButton" name="verify_flash">
              <property name="text">
               <string>Verify</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_5">
           <attribute name="title">
            <string>Fuse</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QTableWidget" name="fuse_table">
              <property name="verticalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOn</enum>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOn</enum>
              </property>
              <property name="showGrid">
               <bool>true</bool>
              </property>
              <property name="rowCount">
               <number>9</number>
              </property>
              <row>
               <property name="text">
                <string>7</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>6</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>5</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>4</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>3</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>2</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>1</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>0</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Value</string>
               </property>
              </row>
              <column>
               <property name="text">
                <string>Low</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>High</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Extended</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="fuse_default">
                <property name="text">
                 <string>Default</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="fuse_read">
                <property name="text">
                 <string>Read</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="fuse_write">
                <property name="text">
                 <string>Write</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_6">
           <attribute name="title">
            <string>EEPROM</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0" colspan="4">
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>EEPROM file:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="eeprom_file"/>
              </item>
              <item>
               <widget class="QComboBox" name="eeprom_format"/>
              </item>
              <item>
               <widget class="QPushButton" name="select_eeprom">
                <property name="text">
                 <string>Select</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="upload_eeprom">
              <property name="text">
               <string>Upload</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QPushButton" name="read_eeprom">
              <property name="text">
               <string>Read</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QPushButton" name="save_eeprom">
              <property name="text">
               <string>Save to...</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QPushButton" name="verify_eeprom">
              <property name="text">
               <string>Verify</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>180</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_7">
           <attribute name="title">
            <string>Lock</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QTableWidget" name="lock_table">
              <property name="verticalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOn</enum>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOn</enum>
              </property>
              <property name="showGrid">
               <bool>true</bool>
              </property>
              <property name="rowCount">
               <number>9</number>
              </property>
              <row>
               <property name="text">
                <string>7</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>6</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>5</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>4</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>3</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>2</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>1</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>0</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeft|AlignTop</set>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Value</string>
               </property>
              </row>
              <column>
               <property name="text">
                <string>Lock Bits</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="lock_default">
                <property name="text">
                 <string>Default</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="lock_read">
                <property name="text">
                 <string>Read</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="lock_write">
                <property name="text">
                 <string>Write</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QPlainTextEdit" name="flash_output">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Settings</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0" colspan="2">
        <layout class="QFormLayout" name="formLayout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Command:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="avrdude_command">
           <property name="text">
            <string>avrdude</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Config file:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Programmer type:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="programmer_type">
           <property name="editable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Programmer port:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="programmer_port">
           <property name="text">
            <string>/dev/parport0</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Programmer speed:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="programmer_speed">
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="currentIndex">
            <number>3</number>
           </property>
           <item>
            <property name="text">
             <string>2400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4800</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>19200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>38400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>115200</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Additional arguments: </string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="avrdude_args"/>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="config_file"/>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>267</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>515</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="save_settings">
         <property name="text">
          <string>Save settings</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>About</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTextBrowser" name="about_view">
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:7pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:18px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
